async-context/README.md

```diff
@@ -0,0 +1,360 @@
+
+```
+---------------------------                    -----------                            
+|   setTimeout()          |                              |
+---------------------------                              |
+|   function f1()         |                              |
+---------------------------                              \/
+|    ...host code...      |                  Continuation Invocation ci1
+---------------------------                              /\  
+|    ...host code...      |                              |
+===================================                      |
+|    Host Continuation A()        |                      |
+===================================            -----------
+```
+
+and
+
+```
+---------------------------                    -----------
+| function x()            |                              |
+---------------------------                              \/                   
+| function myLoggingAPI() |                  Continuation Invocation ci3                        
+===================================                      /\
+|    Continuation f2()            |                      |
+===================================            -----------
+|    ...host code...      |                              |
+---------------------------                              \/
+|    ...host code...      |                   Continuation Invocation ci2                   
+===================================                      /\
+|   Host Continuation B()         |                      |
+===================================            -----------
+```
+
+## Link Context
+We define the `Link Context` `link-c` of a `Continuation` `c` as a pointer from `c` to the `Continuation Invocation` instance where `c` was constructed.  In the example from above, the `continuation` `f2` has a `link context` pointing to `Continuation Invocation` `ci1`.
+
+## Ready Context
+Consider the following example:
+
+```javascript
+const p = new Promise((resolve, reject) => {
+    setTimeout(function f1() {
+       p.then(function f2() {
+             myLoggingAPI('hello');
+         });
+     }, 1000);
+
+     setTimeout(function f3() {
+         resolve(true);
+     }, 2000)
+});
+```
+
+In this code example, we have:
+  - `setTimeout` and `Promise.then` as our `continuation points`. 
+  - four `continuations` - `f1`, `f2` and `f3`, plus the "root continuation" that 
+    the initial code is executing in. 
+  - four `Continuation Invocations`.
+
+We define the `Ready Context` (//TODO, close on naming - `Causal Context`, `Resolve Context`???) for a promise as a reference from a `Continuation Invocation` associated with a Promise.then call, to the `Continuation Invocation` in which the preceding promise in the promise chain was resolved.
+
+For non-promise-based APIs, the `Ready Context` is the same as the `Link Context` (//TODO or should it be null/undefined?)
```

Comment:
My two cents -- there is no need for it to be `null/undefined`.

Reply:
was thinking that the "ready context" can be generalized to the current context when specific state changes happen on a `Continuation Invocation`.  This seems straightforward, and it's a nice definition independent of promises.

Reply:
changed this to be context when the Continuation Invocation's state changes to ready.